### PR TITLE
Fix parameters order during initialisation

### PR DESCRIPTION
**Pull request by SSL32081**
As explained in this [GitHub issue](https://github.com/jax-ml/jax/issues/4085#issuecomment-1186908662), most `jax.tree` functions will shuffle the order of dictionary keys to alphabetical order.

This will cause unexpected behaviours if `dict.values()` (or similar functions) is then called upon the new dictionary, and the order of which is assumed to match that before the `tree` function, ***e.g.*** in [here](https://github.com/kazewong/jim/blob/90a6d519c937fd2e9176f9f358988cafdb71dc1b/). 

This PR address the issue by creating the initial points with list comprehension (in favour of the old plan), skipping the dictionary and the problematic tree.leaves entirely. 

**The old plan**
This PR aims to address this issue, except for places where the outcome is immediately followed by summation or determinants (`jnp.sum`, `jnp.det`, `jnp.reduce`, etc), in which case the ordering does not matter.

### CODE REVIEW

**Comment by kazewong** [comment]
This is addressed in #214. Closing due to duplication
